src/listener_message.rs

```diff
@@ -15,18 +15,11 @@
 // Please review the Licences for the specific language governing permissions and limitations
 // relating to use of the SAFE Network Software.
 
-use sodiumoxide::crypto::sign::PublicKey;
 use socket_addr::SocketAddr;
 
+pub const REQUEST_MAGIC_CONSTANT: [u8; 4] = ['E' as u8, 'C' as u8, 'H' as u8, 'O' as u8];
```

Comment:
> I wonder if this is OK, would it not need `byteorder` also if we are to encrypt all messages very soon, which may require that all get wrapped in an enclosing enum then this may make it harder?

Can you elaborate on this? Maybe split your concerns into different sentences?

About `byteorder`, we only do byte order conversion when we have integers larger than 1 byte. This isn't the case here. Also, [we had code like this before](https://github.com/maidsafe/crust/blob/0.6.1/src/beacon.rs#L36).